app/src/components/ScheduleBuilder.tsx

```diff
@@ -239,13 +238,7 @@ export default class ScheduleBuilder extends Component<Props, States> {
                 <main className="flex flex-row overflow-y-hidden h-full p-1">
                     {/* Picker */}
                     <div className="overflow-y-auto w-full">
-                        <SectionPicker
-                            soc={this.state.soc}
-                            searchText={this.state.searchText}
-                            setSearchText={(searchText) => {
-                                this.setState.bind(this)({ searchText });
-                            }}
-                        />
+                        <SectionPicker soc={this.state.soc} />
```

Comment:
You've got no idea how much nicer this is.

I've always felt that the role of `SectionPicker` should  **strictly** be to show the schedule-of-courses.

The only communication across components should be drag-and-dropping sections & courses.